### PR TITLE
router plugin: HttpAsyncClient xDS router

### LINKED ISSUE #1604: Support obtaining Istio routing and load balancing configurations based on the xDS protocol
# Value
Sermant is a Proxyless Service Mesh Solution with better performance.The support for the xDS protocol enables Sermant to better integrate into the Cloud Native Ecosystem. In Istio, Sermant can replace Envoy as the data plane to obtain directly the configuration of the control plane, and avoid performance loss and architecture complexity caused by Sidecar containers.
# Objective
Sermant can connect to the control plane of Istio through the xDS API interface to obtain the configurations of CDS, EDS, LDS, and RDS, and parse configurations to data model of the Sermant. Sermant framework and plugins will  implement gradually capabilities such as service discovery、routing configuration and load balancing configuration, and the the host microservice can implements the Proxyless Mesh capability by mounting Sermant to  invoke directly the upstream service registered with the control plane, such as istiod.
# RoadMap
**v2.0.0 (2024/06/30)**
Support xDS protocol for service discovery in Sermant framework
**v2.1.0 (2024/09/30)**
Support xDS protocol for routing configuration and load balancing configuration
**v2.2.0 (2024/12/30)**
Support xDS protocol for flow control configuration

### PR DESCRIPTION
**Pull request by daizhenyu**
**What type of PR is this?**

Feature.

**What this PR does / why we need it?**

router plugin: supporting HttpAsyncClient xDS routing

**Which issue(s) this PR fixes？**

Fixes #1604

**Does this PR introduce a user-facing change?**

No

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/sermant-io/Sermant/issues) related with this PR before you start working on it.
- [x] Make sure you have squashed your change to one single commit.
- [ ] GitHub Actions works fine in this PR.

### CODE REVIEW

File: sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/entity/HttpAsyncRequestProducerDecorator.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2022-2022 Huawei Technologies Co., Ltd. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
```
Comment (lilai23) [inline sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/entity/HttpAsyncRequestProducerDecorator.java]: update copyright

File: sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/interceptor/HttpAsyncClient4xInterceptor.java [hunk 2]
```diff
@@ -100,4 +127,60 @@ public ExecuteContext onThrow(ExecuteContext context) {
...
+    }
+
+    private HttpHost rebuildHttpHost(URI newUri) {
+        return URIUtils.extractHost(newUri);
+    }
+
+    private HttpRequest updateRequest(URI newUri, HttpRequest httpUriRequest) {
+        HttpRequestBase httpRequest = (HttpRequestBase) httpUriRequest;
+        httpRequest.setURI(newUri);
+        return httpRequest;
```
Comment (hanbingleixue) [inline sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/interceptor/HttpAsyncClient4xInterceptor.java]: UpdateUri

File: sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/interceptor/HttpAsyncClient4xInterceptor.java [hunk 3]
```diff
@@ -100,4 +127,60 @@ public ExecuteContext onThrow(ExecuteContext context) {
...
+        }
+        URI uri = URI.create(httpRequest.getRequestLine().getUri());
+        String host = uri.getHost();
+        if (!BaseHttpRouterUtils.isXdsRouteRequired(host)) {
+            return;
+        }
+
+        // use xds route to find a service instance, and modify url by it
+        Optional<ServiceInstance> serviceInstanceOptional = BaseHttpRouterUtils
+                .chooseServiceInstanceByXds(host.split("\\.")[0], uri.getPath(), getHeaders(httpRequest));
```
Comment (hanbingleixue) [inline sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/interceptor/HttpAsyncClient4xInterceptor.java]: \\. It is recommended that the value be extracted as a constant.

File: sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/entity/HttpAsyncRequestProducerDecorator.java (hunk 1 above)
Comment (daizhenyu) [reply sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/entity/HttpAsyncRequestProducerDecorator.java]: done

File: sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/interceptor/HttpAsyncClient4xInterceptor.java (hunk 2 above)
Comment (daizhenyu) [reply sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/interceptor/HttpAsyncClient4xInterceptor.java]: updateRequest is clear

File: sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/interceptor/HttpAsyncClient4xInterceptor.java (hunk 3 above)
Comment (daizhenyu) [reply sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/interceptor/HttpAsyncClient4xInterceptor.java]: done

File: sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/interceptor/HttpAsyncClient4xInterceptor.java (hunk 2 above)
Comment (lilai23) [reply sermant-plugins/sermant-router/spring-router-plugin/src/main/java/io/sermant/router/spring/interceptor/HttpAsyncClient4xInterceptor.java]: updateRequestURl？

**Review by hanbingleixue — DISMISSED** [review]

**Review by lilai23 — APPROVED** [review]

**Review by hanbingleixue — APPROVED** [review]

**Pull request merged**